src/main/java/de/maxhenkel/audioplayer/command/AudioPlayerCommands.java

```diff
@@ -417,7 +417,7 @@ private static void processShulker(CommandContext<CommandSourceStack> context, I
             ItemStack itemStack = ItemStack.of(currentItem);
             if (itemValidator.test(itemStack)) {
                 renameItem(context, itemStack, soundID, name);
-                shulkerContents.getCompound(i).put("tag", itemStack.getOrCreateTag());
+                currentItem.put("tag", itemStack.getOrCreateTag());
```

Comment:
Tried this out, but it causes loss of the `Slot` value because that does not exist on `itemStack`. Captured the data for `shulkerContents` below, along with visual result in-game.

**Before:**
```
[
  {
    Count: 1b,
    Slot: 0b,
    id: "minecraft:music_disc_cat"
  },
  {
    Count: 1b,
    Slot: 2b,
    id: "minecraft:goat_horn",
    tag: {
      instrument: "minecraft:yearn_goat_horn"
    }
  },
  {
    Count: 1b,
    Slot: 3b,
    id: "minecraft:jukebox"
  },
  {
    Count: 1b,
    Slot: 5b,
    id: "minecraft:goat_horn",
    tag: {
      instrument: "minecraft:ponder_goat_horn"
    }
  },
  {
    Count: 1b,
    Slot: 13b,
    id: "minecraft:music_disc_cat"
  },
  {
    Count: 1b,
    Slot: 26b,
    id: "minecraft:music_disc_cat"
  }
]
```

**After:**
```
[
  {
    Count: 1b,
    id: "minecraft:music_disc_cat",
    tag: {
      CustomSound: [
        I;1713632501,
        -1539814699,
        -1976373064,
        1817870909
      ],
      HideFlags: 32,
      display: {
        Lore: [
          '{"italic":false,"color":"gray","text":"updatedName"}'
        ]
      }
    }
  },
  {
    Count: 1b,
    Slot: 2b,
    id: "minecraft:goat_horn",
    tag: {
      instrument: "minecraft:yearn_goat_horn"
    }
  },
  {
    Count: 1b,
    Slot: 3b,
    id: "minecraft:jukebox"
  },
  {
    Count: 1b,
    Slot: 5b,
    id: "minecraft:goat_horn",
    tag: {
      instrument: "minecraft:ponder_goat_horn"
    }
  },
  {
    Count: 1b,
    id: "minecraft:music_disc_cat",
    tag: {
      CustomSound: [
        I;1713632501,
        -1539814699,
        -1976373064,
        1817870909
      ],
      HideFlags: 32,
      display: {
        Lore: [
          '{"italic":false,"color":"gray","text":"updatedName"}'
        ]
      }
    }
  },
  {
    Count: 1b,
    id: "minecraft:music_disc_cat",
    tag: {
      CustomSound: [
        I;1713632501,
        -1539814699,
        -1976373064,
        1817870909
      ],
      HideFlags: 32,
      display: {
        Lore: [
          '{"italic":false,"color":"gray","text":"updatedName"}'
        ]
      }
    }
  }
]
```

Reply:
**Before:**
![Before](https://user-images.githubusercontent.com/80792469/229259148-2d27c1ec-0847-4135-bf22-9f4903249a10.png)

**After:**
![After](https://user-images.githubusercontent.com/80792469/229259156-80c8593f-60c4-452d-9e00-04f330dbe82a.png)

Reply:
Oh, my bad, that was an oversight by me. I guess your solution is the best then.